content/client-server-api/_index.md

```diff
@@ -1831,6 +1841,198 @@ the topic to be removed from the room.
+          "count": 3
+        },
+        {
+          "key": "👎",
+          "origin_server_ts": 1562763768320,
+          "count": 1
+        }
+      ],
+      "org.example.possible_thread": {
+        "current_server_participated": true,
+        "count": 7,
+        "latest_event": {
+          "event_id": "$another_event",
+          "content": {
+            "body": "Hello world"
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+Note how the `org.example.possible_annotations` bundle is an array compared to the
+`org.example.possible_thread` bundle where the server is summarising the state of
+the relationship in a single object. Both are valid bundles, and their exact types
+depend on the relationship type.
+
+The endpoints where the server *should* include the `m.relations` unsigned field are:
+
+* [`GET /rooms/{roomId}/messages`](#get_matrixclientv3roomsroomidmessages)
+* [`GET /rooms/{roomId}/context/{eventId}`](#get_matrixclientv3roomsroomidcontexteventid)
+* [`GET /rooms/{roomId}/event/{eventId}`](#get_matrixclientv3roomsroomideventeventid)
+* [`GET /rooms/{roomId}/relations`](#get_matrixclientv3roomsroomidrelations)
+* [`GET /sync`](#get_matrixclientv3sync) when the relevant section has a `limited` value
+  of `true`.
+* [`POST /search`](#post_matrixclientv3search) for any matching events under `room_events`.
+
+{{% boxes/note %}}
+The server is **not** required to return bundles/aggregations on deprecated endpoints
+such as `/initialSync`.
+{{% /boxes/note %}}
+
+While this functionality allows the client to see what was known to the server at the
+time of handling, the client should continue to aggregate locally if it is aware of
+the relationship type's behaviour. For example, a client might increment a `count`
+on a related event's bundle if it saw a new event which referenced that event.
+
+{{% boxes/warning %}}
+The bundle provided by the server only includes events which were known at the time
+the event was *received*. This can mean that in a single `/sync` response an event
+will have a bundle and more events which qualify for that aggregation: in this case,
+the client *should* aggregate the events which are "after" the event in question on
+its own, as the server will not have considered them.
+{{% /boxes/warning %}}
+
+{{% boxes/note %}}
+Events from [ignored users](#ignoring-users) do not appear in the bundle or aggregation.
+Clients will need to de-aggregate the events sent by ignored users to avoid them being
+considered in counts. Similarly, servers must ensure they do not consider events from
+ignored users when preparing a bundle for the client.
+{{% /boxes/note %}}
+
+When an event is redacted, the relations attached to it remain. However, when an event
+which uses a relation is redacted then the relation is broken. Thus, the server needs
+to de-aggregate or disassociate an event from its parent when it is redacted. Clients
+with local aggregation should do the same.
```

Comment:
What does "relations attached to it" mean? I think it means that if you're redacting the parent event it does not affect relations targeting that event.

As opposed to if you redact an event which has an `m.relates_to`.

(While working on the relations code it has become apparent it is very confusing to say "an event with a relation" or "related event" or similar things -- I think it would really help clarify things if the spec were to use defined terms of "parent" and "child" or "target" and "source" or some other terms that are they consistently used throughout.)

Reply:
have changed the dynamic a bit to make it more clear, using the parent/child terminology to clear things up - thanks for the suggestion :)